### DISCUSSION TITLE
Migration of pgsql django.db.utils.ProgrammingError: permission denied for table django_migrations

### DISCUSSION THREAD

**Discussion by atdcode** [netbox-community/netbox#11148] [Q&A]
### NetBox version

3.3.9

### Python version

3.9

### Steps to Reproduce

1. After a fresh install of netbox 3.3.9 on debian 11 with pgsql (14.6-1.pgdg110+1)
2. sudo -u postgres psql -c 'drop database netbox'
3. sudo -u postgres psql -c 'drop database netbox'
4. sudo -u postgres psql -c 'grant all privileges on database netbox to netbox
5. sudo -u postgres psql netbox < my_pgsql14_netbox.sql
6. source /opt/netbox/venv/bin/activate
7. cd /opt/netbox/netbox
8. python3 [manage.py](https://manage.py/) migrate
9. django.db.utils.ProgrammingError: permission denied for table django_migrations

### Expected Behavior

a clean import 

### Observed Behavior

Trying to import a pgsql 14 database in a fresh install of netbox.

**Comment by atdcode** [comment]
Hi 

I forgot to mention i changed the secret_key accordingly to the previous instance of netbox.

Thank you in advance for any help.

**Comment by atdcode** [comment]
So i tried : 

```
sudo -u postgres psql -c 'drop database netbox'
sudo -u postgres psql -c 'create database netbox'
systemctl stop netbox netbox-rq
sudo -u postgres psql -c 'grant all privileges on database netbox to netbox'
sudo -u postgres psql netbox -c "GRANT ALL ON ALL FUNCTIONS IN SCHEMA public to netbox;"
sudo -u postgres psql netbox -c "GRANT ALL ON ALL SEQUENCES IN SCHEMA public to netbox;"

sudo -u postgres psql -c 'GRANT USAGE, SELECT ON ALL SEQUENCES IN SCHEMA public TO netbox;

sudo -u postgres psql -c 'ALTER DATABASE netbox OWNER TO netbox;'

sudo -u postgres psql netbox < /root/atd.new.psql14

source /opt/netbox/venv/bin/activate

cd /opt/netbox/netbox

python3 manage.py migrate
```

i got :

```
Traceback (most recent call last):                                                                                                                                                                                                                                                                                  
  File "/opt/netbox-3.3.9/venv/lib/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute                                                                                                                                                                                                       
    return self.cursor.execute(sql, params)                                                                                                                                                                                                                                                                         
psycopg2.errors.InsufficientPrivilege: permission denied for table django_migrations 
```

Can someone give me a hint please ?

**Comment by cpjones44** [comment]
I got the same error on a fresh install.

Debian 11
Netbox 3.4.3
Postgres 15.1
Python 3.9

Running this worked for me.

sudo -u postgres psql
GRANT postgres TO netbox

**Reply by candlerb** [reply]
That looks like a pretty terrible idea: giving access to postgres internals to the netbox user.

Most likely what you want is:

```
ALTER DATABASE netbox OWNER TO netbox;
```

**Reply by danie-dejager** [reply]
I only got my new install working with the backup after I also ran these commands:
```
sudo -u postgres psql netbox -c "GRANT ALL ON ALL TABLES IN SCHEMA public to netbox;"
sudo -u postgres psql netbox -c "GRANT ALL ON ALL SEQUENCES IN SCHEMA public to netbox;"
sudo -u postgres psql netbox -c "GRANT ALL ON ALL FUNCTIONS IN SCHEMA public to netbox;"
```
Why are they not listed as part of the `replicating netbox` section? 
https://docs.netbox.dev/en/stable/administration/replicating-netbox/

**Reply by candlerb** [reply]
https://docs.netbox.dev/en/stable/installation/1-postgresql/#database-creation

```
ALTER DATABASE netbox OWNER TO netbox;
-- the next two commands are needed on PostgreSQL 15 and later
\connect netbox;
GRANT CREATE ON SCHEMA public TO netbox;
```

It wouldn't be a bad idea to include these steps in the "replicating netbox" section though.
